src/polychord/clustering.f90

```diff
@@ -308,19 +308,19 @@ function cluster(points) result(cluster_list)
             if(nlive>2) then 
                 ! get the position matrix for this cluster
                 if(present(sub_dimensions)) then
-                    points(:nDims, :nlive) =&
-                        RTI%live(sub_dimensions,:nlive,i_cluster)
+                    points(:nlive, :nDims) =&
+                        transpose(RTI%live(sub_dimensions,:nlive,i_cluster))
                 else 
-                    points(:nDims, :nlive) =&
-                        RTI%live(settings%h0:settings%h1,:nlive,i_cluster)
+                    points(:nlive, :nDims) =&
+                        transpose(RTI%live(settings%h0:settings%h1,:nlive,i_cluster))
```

Comment:
RTI provides points the "wrong" way around, so I've transposed it here. Should this be swapped around too?

Reply:
So to be clear, I would advocate having the fortran side of things nDims x nPoints, and the C/Python being nPoints x nDims. For a fortran/C programmer used to converting this is the 'least surprising' way round, and I have a 70% suspicion that if you do it that way round it will end up needing no transposing at the C/Python end.